File: 백준/Silver/28278. 스택 2/스택 2.py
import sys

a = int(sys.stdin.readline())
stack = []

for i in range(a):
    d = list(map(int, sys.stdin.readline().split()))
    
    if d[0] == 1:
        stack.append(d[1])
    elif d[0] == 2:
        if len(stack) >= 1:
            print(stack.pop())
        else:
            print(-1)
    elif d[0] == 3:
        print(len(stack))
    elif d[0] == 4:
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    elif d[0] == 5:
        if len(stack) > 0:
            print(stack[-1])
        else:
            print(-1) 

File: 백준/Silver/28278. 스택 2/스택 2.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        ArrayList<Integer> stack = new ArrayList();
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < n; i++) {
            String a[] = br.readLine().split(" ");

            switch (Integer.parseInt(a[0])) {
                case 1:
                    stack.add(Integer.parseInt(a[1]));
                    break;
                case 2:
                    if (stack.size() >= 1)
                        sb.append(stack.remove(stack.size() - 1)).append("\n");
                    else
                        sb.append(-1).append("\n");
                    break;
                case 3:
                    sb.append(stack.size()).append("\n");
                    break;
                case 4:
                    if (stack.size() == 0) {
                        sb.append(1).append("\n");
                    } else {
                        sb.append(0).append("\n");
                    }
                    break;
                case 5:
                    if (stack.size() > 0) {
                        sb.append(stack.get(stack.size() - 1)).append("\n");
                    } else {
                        sb.append(-1).append("\n");
                    }
            }  
        }
        br.close();
        System.out.println(sb);
    }
}
